app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -242,32 +265,39 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
 
         @Override
         public void onServicesDiscovered(BluetoothGatt gatt, int status) {
-            if (status == BluetoothGatt.GATT_SUCCESS) {
-                BluetoothGattService gattService = mBluetoothGatt.getService(xDripDataService);
-                if (gattService != null) {
-                    BluetoothGattCharacteristic gattCharacteristic = gattService.getCharacteristic(xDripDataCharacteristic);
-                    if (gattCharacteristic != null) {
-                        mCharacteristic = gattCharacteristic;
-                        final int charaProp = gattCharacteristic.getProperties();
-                        if ((charaProp | BluetoothGattCharacteristic.PROPERTY_NOTIFY) > 0) {
-                            mBluetoothGatt.setCharacteristicNotification(gattCharacteristic, true);
-                        } else {
-                            Log.w(TAG, "onServicesDiscovered: characteristic " + xDripDataCharacteristic + " not found");
-                        }
-                    } else {
-                        Log.w(TAG, "onServicesDiscovered: service " + xDripDataService + " not found");
-                    }
-                    Log.d(TAG, "onServicesDiscovered received success: " + status);
-                }
-            } else {
+            if (status != BluetoothGatt.GATT_SUCCESS) {
                 Log.d(TAG, "onServicesDiscovered received: " + status);
+                return;
+            }
+
+            Log.d(TAG, "onServicesDiscovered received status: " + status);
+
+            final BluetoothGattService gattService = mBluetoothGatt.getService(xDripDataService);
+            if (gattService == null) {
```

Comment:
Ah I get it now, so if its not finding your device list some logs so you can see what your characteristics are, makes sense!